src/ftest.jl

```diff
@@ -0,0 +1,140 @@
+type FTestResult{N}
+    ssr::NTuple{N, Float64}
+    dof::NTuple{N, Int}
+    dof_resid::NTuple{N, Int}
+    r2::NTuple{N, Float64}
+    fstat::Tuple{Vararg{Float64}}
+    pval::Tuple{Vararg{PValue}}
+end
+
+"""A helper function to determine if mod1 is nested in mod2"""
+function issubmodel(mod1::LinPredModel, mod2::LinPredModel)
+    mod1.rr.y != mod2.rr.y && return false # Response variables must be equal
+
+    # Now, test that all predictor variables are equal
+    pred1 = mod1.pp.X
+    npreds1 = size(pred1, 2)
+    pred2 = mod2.pp.X
+    npreds2 = size(pred1, 2)
+    # If model 1 has more predictors, it can't possibly be a submodel
+    npreds1 > npreds2 && return false 
+
+    @inbounds for i in 1:npreds1
+        var_in_mod2 = false
+        for j in 1:npreds2
+            if view(pred1, :, i) == view(pred2, :, j)
+                var_in_mod2 = true
+                break
+            end
+        end
+
+        if !var_in_mod2 # We have found a predictor variable in model 1 that is not in model 2
+            return false 
+        end
+    end
+
+    return true
+end
+
+_diffn{N, T}(t::NTuple{N, T})::NTuple{N, T} =  ntuple(i->t[i]-t[i+1], N-1)
```

Comment:
`::NTuple{N, T}` is actually wrong here, since the result has `N-1` elements. Just drop the type assertion (same below).